packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/next/MediaLibraryNextModule.kt

```diff
@@ -0,0 +1,175 @@
+        self.assets
+      }
+
+      AsyncFunction("add") Coroutine { self: Album, asset: Asset ->
+        systemPermissionsDelegate.requireSystemPermissions(true)
+        mediaStorePermissionsDelegate.requestMediaLibraryActionPermission(listOf(asset.contentUri))
+        self.add(asset)
+      }
+
+      AsyncFunction("delete") Coroutine { self: Album ->
+        systemPermissionsDelegate.requireSystemPermissions(true)
+        val assetIdsToDelete = self.assets.map { it.contentUri }
+        mediaStorePermissionsDelegate.requestMediaLibraryActionPermission(assetIdsToDelete, needsDeletePermission = true)
+        self.delete()
+      }
+    }
+
+    AsyncFunction("createAsset") Coroutine { filePath: String, album: Album? ->
+      systemPermissionsDelegate.requireSystemPermissions(false)
+      val assetFactory = AssetFactory(context)
+      return@Coroutine assetFactory.create(filePath, album?.getRelativePath())
+    }
+
+    @OptIn(EitherType::class)
+    AsyncFunction("createAlbum") Coroutine { name: String, assetsRefs: Either<List<Asset>, List<String>>, move: Boolean ->
+      systemPermissionsDelegate.requireSystemPermissions(false)
+      val albumFactory = AlbumFactory(context)
+      if (assetsRefs.`is`(toKClass<List<Asset>>())) {
+        assetsRefs.get(toKClass<List<Asset>>()).let {
+          return@Coroutine albumFactory.createFromAssets(name, it, move)
+        }
+      }
+      if (assetsRefs.`is`(toKClass<List<String>>())) {
+        assetsRefs.get(toKClass<List<String>>()).let {
+          return@Coroutine albumFactory.createFromFilePaths(name, it, AssetFactory(context))
+        }
+      }
+    }
+
+    AsyncFunction("deleteManyAlbums") Coroutine { albums: List<Album> ->
+      systemPermissionsDelegate.requireSystemPermissions(true)
+      albums.forEach { album -> album.delete() }
+    }
+
+    AsyncFunction("deleteManyAssets") Coroutine { assets: List<Asset> ->
+      systemPermissionsDelegate.requireSystemPermissions(true)
+      val assetIdsToDelete = assets.map { it.contentUri }
+      mediaStorePermissionsDelegate.requestMediaLibraryActionPermission(assetIdsToDelete, needsDeletePermission = true)
+      assets.forEach { asset -> asset.delete() }
+    }
+
+    AsyncFunction("requestPermissionsAsync") { writeOnly: Boolean, permissions: List<GranularPermission>?, promise: Promise ->
+      systemPermissionsDelegate.requestPermissions(writeOnly, permissions, promise)
+    }
+
+    AsyncFunction("getPermissionsAsync") { writeOnly: Boolean, permissions: List<GranularPermission>?, promise: Promise ->
+      systemPermissionsDelegate.getPermissions(writeOnly, permissions, promise)
+    }
+
+    RegisterActivityContracts {
+      mediaStorePermissionsDelegate.deleteLauncher =
+        registerForActivityResult(DeleteContract(this@MediaLibraryNextModule))
+      mediaStorePermissionsDelegate.writeLauncher =
+        registerForActivityResult(WriteContract(this@MediaLibraryNextModule))
```

Comment:
```suggestion
      mediaStorePermissionsDelegate.registerContracts(this@MediaLibraryNextModule)
```
I think moving it to the delegate is a good idea.